### ISSUE TITLE
macOS version not reported

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dotnet#2710]
example event: https://sentry-sdks.sentry.io/issues/4536804443/?project=5428537&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=1h&stream_index=1

![image](https://github.com/getsentry/sentry-dotnet/assets/1633368/4ab8d977-4efa-4a3f-ad4f-4584fd39f714)

Seems like macOS version has regressed

**Comment by jamescrosswell** [comment]
We are still collecting the OS version however it's stored in the Contexts (not in the Tags):
https://github.com/getsentry/sentry-dotnet/blob/5bcb9cac8b54f283a4ce5cc16573bfdf2970cdf9/src/Sentry/Internal/Enricher.cs#L49

This is what I'm seeing come through to Sentry:
> **Operating System**
> _Name_ Darwin
> _Kernel Version_ 23.4.0
> _Raw Description_ Darwin 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:49 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6020

It gives us the Darwin version, which is a bit obtuse. We could potentially try to pretty this up in the case of macOS by [doing a bit of a lookup](https://en.wikipedia.org/wiki/MacOS_Sonoma#Release_history) to work out the common name and version (e.g. Sonoma 14.4.1), but I believe we'd have to maintain that lookup table manually everytime Apple released a new macOS version.

**Comment by bruno-garcia** [comment]
Relay extract tags from the context. We shouldn't be sending any custom tag directly from the SDK.

**Comment by jamescrosswell** [comment]
> Relay extract tags from the context. We shouldn't be sending any custom tag directly from the SDK.

Ok, so currently the behaviour is slightly different for macOS than Windows. 

On Windows you end up with tags for `os` and `os.name`:
<img width="639" alt="image" src="https://github.com/getsentry/sentry-dotnet/assets/728212/9ebe3952-4d87-4b63-bd06-3335f832504d">

On macOS you end up with just the `os.name`:
<img width="293" alt="image" src="https://github.com/getsentry/sentry-dotnet/assets/728212/1ca6614d-5750-4d4e-a9ad-0ef8954f0fe0">

What is different on MacOS is that we provide the `Kernel Version` in the context, whereas on Windows it's `Version`:
<img width="972" alt="image" src="https://github.com/getsentry/sentry-dotnet/assets/728212/2907315e-0a35-4609-8e83-84b547cd9212">
<img width="978" alt="image" src="https://github.com/getsentry/sentry-dotnet/assets/728212/4be17a3a-4672-41dc-98fb-d498377cea74">

That's maybe what's tripping up the relay. I think it [only looks for a value for `os.version`](https://github.com/getsentry/relay/blob/24037f40afe892bf75bba3e74c4fc10c7f710bb1/relay-event-normalization/src/normalize/contexts.rs#L177-L179) in the context.

Do we want to try to fix in the SDK or in the Relay? It might be nicer to do it in the relay, since otherwise we'd probably have to change multiple SDKs?

**Comment by philipphofmann** [comment]
MacOS version works on Cocoa: https://sentry-sdks.sentry.io/issues/5310027106/events/1b44a1146aab4fc98a547b815e22b8da/

![CleanShot 2024-05-06 at 13 19 34](https://github.com/getsentry/sentry-dotnet/assets/2443292/5d44285b-9d63-426a-9ca9-29b21091faeb)

**Comment by bitsandfoxes** [comment]
> Do we want to try to fix in the SDK or in the Relay?

This looks like a job that should be done in Relay.

**Comment by bruno-garcia** [comment]
Would be nice to fix this. If we can do it in Relay, that's great. 
If we need to add `netX-macos` and get the proper OS name/version, we can do that too. The "Darwin" version isn't great.

**Comment by jamescrosswell** [comment]
See @Dav1dde's reply:
- https://github.com/getsentry/relay/issues/4109#issuecomment-2396437066

Looks like we should try to get this from a different source. I'll dig around for a proper API to pull that information rather than trying to read it from system files (that we may not have read access to).

**Comment by bruno-garcia** [comment]
yeah I think we need to report the right data. We might need to add `netX.0-macos` if we need to `#if MAC` and access some native APIs.

Or actually we prob can get away with the `P/Invoke` signature for the function and just call the right thing if we're on a Mac. Like things used to be done before the platform specific TFMs.

Whatever they are calling in the cocoa SDK to get this data, we can call that via P/Invoke when running on a Mac
